### PR TITLE
add some iter type

### PR DESCRIPTION
**Pull request by GG2002**

### CODE REVIEW

File: src/iter.rs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    Ix: IndexType,
+{
+    type Item = (&'a Interval<T>, &'a V);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.stack.is_empty() {
+            return None;
+        }
+        let mut x = self.stack.pop().unwrap();
```
Comment (bsbds) [inline src/iter.rs]: please use `let mut x = self.stack.pop()?;`

File: src/iter.rs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+            .map_ref
+            .node_ref(x, Node::interval)
+            .overlaps(self.query)
+        {
+            self.stack.extend(left_link_with_query(
+                self.map_ref,
+                self.map_ref.node_ref(x, Node::right),
+                self.query,
+            ));
+            if self.stack.is_empty() {
```
Comment (bsbds) [inline src/iter.rs:240]: Same as above.

File: src/iter.rs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    Ix: IndexType,
+{
+    type Item = (&'a Interval<T>, &'a V);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.stack.is_empty() {
+            return None;
+        }
+        let x = self.stack.pop().unwrap();
```
Comment (bsbds) [inline src/iter.rs]: Same as above.

File: src/iter.rs [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    Ix: IndexType,
+{
+    type Item = (Interval<T>, V);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.stack.is_empty() {
+            return None;
+        }
+        let x = self.stack.pop().unwrap();
```
Comment (bsbds) [inline src/iter.rs]: Same as above.

File: benches/bench.rs [hunk 5]
```diff
@@ -1,44 +1,21 @@
...
     }
 
-    fn next(&mut self) -> Interval<i32> {
-        let low = self.rng.gen_range_i32(0, self.limit - 1);
-        let high = self.rng.gen_range_i32(low + 1, self.limit);
+    fn next(&mut self) -> Interval<u32> {
+        let low = self.rng.gen();
+        let high = self.rng.gen_range(low + 1..=u32::MAX);
         Interval::new(low, high)
     }
```
Comment (bsbds) [inline benches/bench.rs]: Why do you remove the `limit`? In this case, the intervals in the benchmark will rarely overlapping.

Comment (GG2002) [reply benches/bench.rs]: Sorry, I overlooked this point.

**Review by bsbds — APPROVED** [review]

**Pull request merged**